sql3/planner/expressionagg.go

```diff
@@ -943,6 +944,311 @@ func (n *percentilePlanExpression) WithChildren(children ...types.PlanExpression
 	return newPercentilePlanExpression(children[0], children[1], n.returnDataType), nil
 }
 
+// aggregator for CORR()
+type aggregateCorr struct {
```

Comment:
I feel like somewhere in here, we could really use a comment explaining what is actually happening. I last took a statistics class something like 33 years ago and I do not actually know what is happening here. Some quick summary of the name of the algorithm or something, and an explanation of what we're tracking and why tracking it is useful seems rewarding.

Reply:
https://lmgtfy.app/?q=correlation+coefficient